feat: TreeSelect support maxCount

aojunhao123 opened the pull request:
### 🤔 This is a ...

- [x] 🆕 New feature

### 🔗 Related Issues
close #51490 

### 💡 Background and Solution

The TreeSelect component currently lacks the ability to limit the maximum number of selections, which is a common requirement in many scenarios. This PR adds this functionality through:
- Adding maxCount prop to TreeSelect component
- Adding demo and documentation

### 📝 Change Log

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 🆕 Add maxCount prop to TreeSelect to limit the maximum number of selections. |
| 🇨🇳 Chinese | 🆕 为 TreeSelect 组件新增 maxCount 属性以限制最大选择数量。|

zombieJ committed:
refactor: Button inner logic & use effect to get focus (#51624)

* chore: init

* test: update test

Wxh16144 committed:
chore: Improve visual regression report (#51633)

zombieJ committed:
fix: Form with multiple same error render error (#51636)

* fix: key warning

* test: add test case

DDDDD12138 committed:
docs: add missing version column and correct version placement (#51635)

aojunhao123 committed:
feat: TreeSelect support maxCount

aojunhao123 commented:
Waiting for the [prerequisite PR](https://github.com/react-component/tree-select/pull/596) to be merged.

thinkasany commented on components/tree-select/index.zh-CN.md:
```diff
@@ -51,6 +52,7 @@ demo:
 | labelInValue | 是否把每个选项的 label 包装到 value 中，会把 value 类型从 `string` 变为 {value: string, label: ReactNode, halfChecked(treeCheckStrictly 时有效): string\[] } 的格式 | boolean | false |  |
 | listHeight | 设置弹窗滚动高度 | number | 256 |  |
 | loadData | 异步加载数据。在过滤时不会调用以防止网络堵塞，可参考 FAQ 获得更多内容 | function(node) | - |  |
+| maxCount | 指定可选中的最多 items 数量，仅在多选时生效 | number | - | 5.23.0 |
```
```suggestion
| maxCount | 指定可选中的最多 items 数量，仅在 `multiple` 设置时生效 | number | - | 5.23.0 |
```

thinkasany replied:
不确定这样子会不会更好，虽然中文语意也挺清晰的，但是感觉突出 `api` 应该没坏处，其他地方好像也是这样子用的。

<img width="1279" alt="image" src="https://github.com/user-attachments/assets/f442f577-d01a-4cbf-87e6-dde0fa374d94">

aojunhao123 committed:
Update components/tree-select/index.zh-CN.md

Co-authored-by: thinkasany <480968828@qq.com>
Signed-off-by: Jony J <1844749591@qq.com>

zombieJ committed:
enhance: not trigger scroll on ActionButton (#51647)

dependabot[bot] committed:
chore(deps): bump codecov/codecov-action from 4 to 5 (#51658)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

afc163 committed:
chore: improve Contributors loading suspense style (#51663)

thecodesalim committed:
fix: prevent OTP input focus from advancing when previous input is empty (#51664)

Co-authored-by: Salim Abubakar <salimabubakar@Salims-MacBook-Air.local>

thinkasany committed:
docs: show suffix demo (#51666)

afc163 committed:
refactor: tweak button style code (#51646)

zombieJ committed:
feat: Modal `width` support responsive size (#51653)

* feat: Modal support responsive width

* docs: update doc

* test: update snapshot

Wxh16144 committed:
style: Improve style z-order to fix some errors (#51448)

Co-authored-by: IsKaros <IsKaros@users.noreply.github.com>
Co-authored-by: ​ <​>

Zyf665 committed:
fix(Table): preserve custom event handlers in row selection (#51661)

Wxh16144 committed:
chore: Update scripts/visual-regression/report-template.html (#51672)

afc163 committed:
fix: Button custom icon is not align center (#51652)

MadCcc committed:
docs: changelog 5.22.2 (#51670)

* docs: changelog 5.22.2

* Apply suggestions from code review

Co-authored-by: afc163 <afc163@gmail.com>
Signed-off-by: MadCcc <1075746765@qq.com>

* Update CHANGELOG.en-US.md

Co-authored-by: thinkasany <480968828@qq.com>
Signed-off-by: MadCcc <1075746765@qq.com>

* docs: update

---------

Signed-off-by: MadCcc <1075746765@qq.com>
Co-authored-by: afc163 <afc163@gmail.com>
Co-authored-by: thinkasany <480968828@qq.com>

zhangchao-wooc committed:
docs: Cascader notFoundContent should be ReactNode (#51684)

LorenzoCardinali committed:
fix: add missed translation in it_IT (#51685)

MadCcc committed:
chore: update changelog date (#51686)

DDDDD12138 committed:
docs(list): remove unexpected message on initial data load (#51687)

afc163 committed:
chore: remove useless var (#51689)

thinkasany committed:
type(TextArea): combine TextAreaRef and TextAreaProps imports (#51695)

MadCcc committed:
docs: update changelog 5.22.2 (#51688)

* chore: update changelog date

* docs: update changelog

* docs: update bugver

* Update CHANGELOG.en-US.md

Co-authored-by: thinkasany <480968828@qq.com>
Signed-off-by: MadCcc <1075746765@qq.com>

---------

Signed-off-by: MadCcc <1075746765@qq.com>
Co-authored-by: thinkasany <480968828@qq.com>

jynxio committed:
fix: Vertical alignment of clear icon in Input (#51700)

zombieJ committed:
fix: Select align issue (#51694)

* fix: item align logic

* fix: placeholder style

* test: update snapshot

* fix: single align

* chore: add demo

* test: update snapshot

* fix: color

* fix: style

* fix: line height

* fix: borderless pos

* chore: base of prefix offset

li-jia-nan committed:
chore: up release data (#51703)

Co-authored-by: 栗嘉男 <lijianan@lijianandeMacBook-Pro.local>

MadCcc committed:
chore: update pre-publish (#51708)

MadCcc committed:
chore: update pre-publish script

MadCcc committed:
chore: update changelog date

MadCcc committed:
chore: update pre-publish

zombieJ committed:
chore: use single update status of diff action

zombieJ committed:
chore: tmp lock `@swc/types` ver (#51722)

* chore: tmp lock types ver

* chore: force lock

985563349 committed:
chore: optimize sort gradient non-numeric judgments (#51720)

Co-authored-by: lijianan <574980606@qq.com>

li-jia-nan committed:
chore: up ChangeLog (#51719)

Jinbao1001 committed:
chore: unlock @swc/types (#51729)

github-actions[bot] committed:
chore: upgrade deps (#51733)

Co-authored-by: afc163 <507615+afc163@users.noreply.github.com>

thinkasany committed:
feat(Segmented): support `name` prop to enable grouping (#51725)

jynxio committed:
docs: Rule.message type definition in Form Comp (#51709)

renovate[bot] committed:
chore(deps): update dependency @dnd-kit/sortable to v9 (#51751)

* chore(deps): update dependency @dnd-kit/sortable to v9

* test: update test snap

* test: update test snap

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: 栗嘉男 <574980606@qq.com>

renovate[bot] committed:
chore(deps): update dependency typescript to ~5.7.0 (#51749)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: lijianan <574980606@qq.com>

afc163 committed:
chore: enable codecov bundle analyze (#51753)

Co-authored-by: lijianan <574980606@qq.com>

dependabot[bot] committed:
chore(deps-dev): bump react-router-dom from 6.28.0 to 7.0.0 in the dev-dependencies group (#51744)

* chore(deps-dev): bump react-router-dom in the dev-dependencies group

Bumps the dev-dependencies group with 1 update: [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom).


Updates `react-router-dom` from 6.28.0 to 7.0.0
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router-dom@7.0.0/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: dev-dependencies
...

Signed-off-by: dependabot[bot] <support@github.com>

* test: fix test case

* fix: fix

* fix: fix rename

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: lijianan <574980606@qq.com>
